### PR TITLE
Transformations: Allow more complex regex expressions in `Rename by regex`

### LINKED ISSUE #46894: Grafana - Rename by Regex doesnt work with global patterns
Hi, I have the following field in my query:

Example: `{method="GET", status="200", uri="/v2/path1/jhkdsahkjdhas89d2182137/path2/3213213dsagg4312321/path3"}`

I want to apply a transformation in Grafana to remove the path parameters from URI so I could properly group them.

But when I apply the `Rename By Regex` transformation, it applies to the first match. I am unable to use the `\g` to set the global flag in the Regex.

My Regex expression: `/(\/([A-Za-z]*[0-9]+)+\/)/g`
My Replace value: `path-param`
The result I expected: `{method="GET", status="200", uri="/v2/path1/path-param/path2/path-param/path3"}`
The result I receive: `{method="GET", status="200", uri="/v2/path1/path-param/path2/3213213dsagg4312321/path3"}`

The only workaround I found is to repeat the transformation which is not feasible since I have multiple endpoints with different numbers of path-params each one.

### PR DESCRIPTION
**Pull request by ashharrison90**
**What this PR does / why we need it**:

- `Rename by regex` is currently inconsistent between the frontend validation and the frontend runtime
  - validation uses `stringToJsRegex` to validate whether it's correct regex, whereas the runtime just does `new Regex(<expression>)`
- switch to using `stringToJsRegex` in the runtime. this is then consistent with the validation and consistent with other areas of grafana (value mappings for example).
- this also has the benefit of allow for more complex expressions in the `match` field

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/46894
Fixes https://github.com/grafana/support-escalations/issues/2440

**Special notes for your reviewer**:

# Release notice breaking change

The `Rename by regex` transformation has been improved to allow global patterns of the form `/<stringToReplace>/g`. Depending on the regex match used, this may cause some transformations to behave slightly differently. You can guarantee the same behaviour as before by wrapping the `match` string in forward slashes (`/`), e.g. `(.*)` would become `/(.*)/`

### CODE REVIEW

File: packages/grafana-data/src/transformations/transformers/renameByRegex.test.ts [hunk 1]
```diff
@@ -147,7 +200,6 @@ describe('Rename By Regex Transformer', () => {
           Array [
             Object {
               "config": Object {
-                "displayName": "Time",
```
Comment (JoaoSilvaGrafana) [inline packages/grafana-data/src/transformations/transformers/renameByRegex.test.ts:150]: Why was this removed? 🤔

**Review by JoaoSilvaGrafana — APPROVED** [review]
Lgtm! Nice job explaining it in the release notes. Just left 1 question

File: packages/grafana-data/src/transformations/transformers/renameByRegex.test.ts (hunk 1 above)
Comment (ashharrison90) [reply packages/grafana-data/src/transformations/transformers/renameByRegex.test.ts:150]: ah ok so it's to do with the subtle differences in the different regexp implementations.

with the old way, `regex.test(displayName)` is truthy so it goes on to do the replace logic (which ultimately does nothing because it's replacing an empty string...): https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/transformations/transformers/renameByRegex.ts#L53

with the new logic, `regex.test(displayName)` is falsey so the code immediately short-circuits and returns the field as it was.

**Comment by zuchka** [comment]
this is awesome. users really love this feature ❤️
